hs_core/hydroshare/resource.py

```diff
@@ -1001,15 +1001,25 @@ def delete_resource_file(pk, filename_or_id, user, delete_logical_file=True):
     resource = utils.get_resource_by_shortkey(pk)
     res_cls = resource.__class__
 
+    def can_delete_logical_file(logical_file):
```

Comment:
Definitely belongs with the logical file, not the resource and separates the logic of the different types.

Reply:
Fixed as suggested